libs/component-library/src/lib/progress-bar/progressBar.tsx

```diff
@@ -17,12 +27,31 @@ const FxProgressBar = ({
   ...rest
 }: FxProgressBarProps) => {
   const [_width, setWidth] = React.useState(width || 0);
-  const _progress = Math.min(Math.max(0, progress), total);
-  const progressWidth = Math.round(_width * (_progress / total));
+  const progressWidth = useSharedValue<number>(0);
 
   React.useEffect(() => {
-    configureEaseInOutLayoutAnimation();
-  }, [progressWidth]);
+    const newWidth = calculateWidth(progress, total, _width);
+    if (newWidth !== progressWidth.value) {
+      progressWidth.value = withTiming(newWidth, {
+        duration: 150,
+      });
+    }
+  }, [progress, total, _width]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const progressWidthStyle = useAnimatedStyle(() => {
+    return {
+      width: progressWidth.value,
+    };
+  });
+
+  const onLayoutHandler = (evt: LayoutChangeEvent) => {
+    setWidth(evt.nativeEvent.layout.width);
+    progressWidth.value = calculateWidth(
+      progress,
+      total,
+      evt.nativeEvent.layout.width
+    );
+  };
```

Comment:
Would this not cause `width` to fire the `useEffect` updating `progressWidth.value` anyway?

Reply:
This is initializing `progressWidth.value` so that the `useEffect` doesn't cause it to do the animation on first render.